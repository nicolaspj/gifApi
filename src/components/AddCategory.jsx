import { useState } from "react"

 

export const AddCategory = ({ onNewCategory }) => {
    const [ inputValue , setInputValue ] = useState( 'One' )    

    const onInputChange = ({ target }) => {
        console.log(target.value);
        setInputValue( target.value );
    }

    const onSubmit = ( event ) => {
         event.preventDefault(); 
         if(inputValue.trim().length <= 1) return; //para que ingrese un caracter y no ingrese vacio 
        // setCategories( categories => [ inputValue , ...categories ] )
        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return (
        <form onSubmit={ onSubmit } >
            <input 
                
                type="text"
                placeholder="Buscar Gifs"
                value={ inputValue }
                onChange={ onInputChange }
            />
        </form>
)
}
