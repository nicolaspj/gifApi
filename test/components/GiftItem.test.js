import { render, screen } from '@testing-library/react'
import { GiftItem } from '../../src/components/GiftItem'


describe('Test GifItem', () => {
    const title = 'DBZ';
    const url = 'https://damndemo.com.ar/favicon.jpg'


    test('Debe enviar title y url + snapshot', () => {


        const { container } = render(<GiftItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })


    test('debe mostrar la imagen con el URL y el ALT inidicado', () => {
        render(<GiftItem title={title} url={url} />)
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(title + "?");


    })

    test('Debe mostrar el titulo en el componente', () => {
        render(<GiftItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy();
    })
})