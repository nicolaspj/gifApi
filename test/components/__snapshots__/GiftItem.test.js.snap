// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem Debe enviar title y url + snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="DBZ"
      src="https://damndemo.com.ar/favicon.jpg"
    />
    <p>
       
      DBZ
       
    </p>
  </div>
</div>
`;
